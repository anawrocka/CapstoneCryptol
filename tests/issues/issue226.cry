module issue226 where

import issue226r2

foo x = x
