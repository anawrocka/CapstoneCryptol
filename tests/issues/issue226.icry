:l issue226.cry
:browse
