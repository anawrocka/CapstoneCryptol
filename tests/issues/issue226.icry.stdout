Loading module Cryptol
Loading module Cryptol
Loading module issue226r2
Loading module issue226
Type Synonyms
=============
    type Bool = Bit
    type Char = [8]
    type String n = [n][8]
    type Word n = [n]

Symbols
=======
    drop : {front, back, elem} (fin front) => [front +
                                               back]elem -> [back]elem
    foo : {a} a -> a
    groupBy : {each, parts, elem} (fin each) => [parts *
                                                 each]elem -> [parts][each]elem
    splitBy : {parts, each, elem} (fin each) => [parts *
                                                 each]elem -> [parts][each]elem
    tail : {a, b} [1 + a]b -> [a]b
    take : {front, back, elem} (fin front) => [front +
                                               back]elem -> [front]elem
    undefined : {a} a
    width : {bits, len, elem} (fin len, fin bits,
                               bits >= width len) => [len]elem -> [bits]

